### PR TITLE
add nwaku guides

### PR DESCRIPTION
**Pull request by LordGhostX**
Todo:

- [x] add guide for building nwaku from the source
- [x] add guide for running nwaku in a docker container
- [x] add guide for running nwaku-compose
- [x] add guide for nwaku configuration methods
- [x] add reference page for nwaku options
- [x] add guides for configuring peer discovery
- [ ] add config examples for nwaku
- [ ] add guide for monitoring nwaku with Prometheus and Grafana
- [ ] add guide for running nwaku in the cloud (DigitalOcean, flywaku)

### CODE REVIEW

File: docs/guides/run-nwaku-node.md [hunk 1]
```diff
@@ -34,17 +38,8 @@ Once you have built the `nwaku` node, run it using the default configuration:
...
-- Enable the `Relay` protocol for relaying messages.
-- Enable the `Store` protocol as a client, allowing it to query peers for historical messages but not store any message itself.
-
-:::info
-For more advanced configurations like enabling other protocols or maintaining a consistent `PeerID`, please refer to the [Configuration Methods](https://github.com/waku-org/nwaku/blob/master/docs/operators/how-to/configure.md) guide.
+:::tip
+For more advanced configurations like enabling other protocols or maintaining a consistent `PeerID`, please refer to the [Node Configuration Methods](/guides/reference/node-config-methods) guide.
 :::
 
 ## Connect the Node
```
Comment (fryorcraken) [inline docs/guides/run-nwaku-node.md:51]: There are two part to this topic:

1. Bootstrapping to the Waku network
2. Discovering more peer

DNS Discovery and static peers solve (1)
Discv5 solves (2)

However, can't solve (2) without solving (1). I think this section should focus on (1) and be called "Bootstrapping" the node or something like that.

Then, second section can be  on "peer discovery" 

Both section having a single liner on what bootstrap and peer discovery are.

File: docs/guides/nwaku/build-source.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
+---
+title: Build Nwaku from Source
+---
+
+This guide provides detailed steps to build a `nwaku` node from the source to access the latest development version or a specific commit or tag of `nwaku`. If you prefer a more stable version, [download a pre-compiled binary](https://github.com/waku-org/nwaku/tags) instead.
```
Comment (fryorcraken) [inline docs/guides/nwaku/build-source.md:5]: You can have a more stable version by compiling the latest releaes.

using pre-compiled binary is more convenient.

File: docs/guides/run-nwaku-node.md [hunk 3]
```diff
@@ -4,27 +4,37 @@ title: Run a Nwaku Node
...
 
-Before running a Nwaku node, it is necessary to build it. Nwaku provides multiple options for building a node:
+To run a node, you must have the `nwaku` binary. Nwaku provides multiple options for acquiring the node binary:
+
+#### Download the Binary
 
 | | Description | Documentation |
 | - | - | - |
-| Source Code | Build a `nwaku` node directly from the source code | [Build Nwaku from Source](https://github.com/waku-org/nwaku/blob/master/docs/operators/how-to/build.md) |
 | Precompiled Binary | Download a precompiled binary of the `nwaku` node | [Download Nwaku Binary](https://github.com/waku-org/nwaku/tags) |
```
Comment (fryorcraken) [inline docs/guides/run-nwaku-node.md:17]: @jm-clius do we sign our binaries?

File: docs/guides/nwaku/build-source.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+```bash
+sudo apt-get install build-essential git
+```
+
+</TabItem>
+<TabItem value="fedora" label="Fedora">
+
+```bash
+dnf install @development-tools
```
Comment (fryorcraken) [inline docs/guides/nwaku/build-source.md:36]:
```suggestion
sudo dnf install @development-tools
```

File: docs/guides/nwaku/build-source.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+```bash
+# Using your favorite AUR helper
+[AUR HELPER] -S base-devel
+```
+
+</TabItem>
+</Tabs>
+
+#### macOS
```
Comment (fryorcraken) [inline docs/guides/nwaku/build-source.md:50]:
```suggestion
#### MacOS
```

Maybe worth putting Mac as a tab with the linux distro? Tab title could be "MacOS (Homebrew)"

File: docs/guides/nwaku/build-source.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+Build the `nwaku` binary:
+
+```bash
+make wakunode2
+```
+
+The first `make` invocation updates to all Git submodules. After each `git pull`, run `make update` to keep the submodules updated in the future.
+
+```bash
+make update wakunode2
```
Comment (fryorcraken) [inline docs/guides/nwaku/build-source.md:82]:
```suggestion
make update
```

I don't think `wakunode2` changes anything. @jm-clius to confirm.

File: docs/guides/nwaku/run-docker.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+---
+
+This guide provides detailed steps to build and run a `nwaku` node in a Docker container. If you prefer a pre-configured setup that includes a monitoring dashboard, see the [Run Nwaku with Docker Compose](/guides/nwaku/run-docker-compose) guide.
+
+## Prerequisites
+
+Ensure [Docker](https://www.docker.com/) is installed on your system using the appropriate instructions provided in the [Docker documentation](https://docs.docker.com/engine/install/). For example, you can use Docker's convenience script for installation:
+
+```bash
+curl -fsSL https://get.docker.com -o get-docker.sh
```
Comment (fryorcraken) [inline docs/guides/nwaku/run-docker.md:12]: Does this work on both MAc and Linux?

File: docs/guides/nwaku/run-docker.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+You can also build the Docker image locally:
+
+```bash
+# Clone the repository
+git clone --recurse-submodules https://github.com/waku-org/nwaku
+cd nwaku
+
+# Build image using docker build
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply docs/guides/nwaku/run-docker.md:33]: Let's make it a tab? I think users only need the `make` command. Or even remove the `docker` one.

File: docs/overview/concepts/static-peers.md [hunk 9]
```diff
@@ -1,15 +1,15 @@
 ---
-title: Predefined Nodes
+title: Static Peers
 ---
 
-Waku applications have the flexibility to embed bootstrap node addresses directly into their codebase. Developers can use either the [predefined nodes by Status](https://github.com/waku-org/js-waku/blob/master/packages/core/src/lib/predefined_bootstrap_nodes.ts#L45) or [run a node](/guides/nodes-and-sdks#run-a-waku-node) per their preference.
+Waku applications have the flexibility to embed bootstrap node addresses directly into their codebase. Developers can either use [static peers by Status](https://github.com/waku-org/js-waku/blob/master/packages/core/src/lib/predefined_bootstrap_nodes.ts#L45) or [run a node](/guides/nodes-and-sdks#run-a-waku-node) per their preference.
```
Comment (fryorcraken) [inline docs/overview/concepts/static-peers.md:5]:
```suggestion
Waku applications have the flexibility to embed bootstrap node addresses directly into their codebase. Developers can either use [static peers operated by Status](https://github.com/waku-org/js-waku/blob/master/packages/core/src/lib/predefined_bootstrap_nodes.ts#L45) or [run a node](/guides/nodes-and-sdks#run-a-waku-node).
```

"per their preference" sounds odd

File: docs/guides/reference/node-config-methods.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
+---
+title: Node Configuration Methods
+---
+
+Waku nodes can be configured using a combination of the following methods:
+
+1. Command line options and flags
+2. Environment variables (recommended)
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply docs/guides/reference/node-config-methods.md:8]: I don't think it's recommended either. It's just preferred by some operator in some contexts.

File: docs/overview/reference/glossary.md [hunk 11]
```diff
@@ -58,6 +58,10 @@ Mostly offline devices refer to clients who spend most of their time offline or
 
 A node is a device or client that implements Waku [protocols](#protocol) and leverages the [Waku Network](#waku) to enable secure and private peer-to-peer Web3 communication.
 
+### Node Key
+
+A node key is a [Secp256k1](https://en.bitcoin.it/wiki/Secp256k1) (64 char hex string) private key for generating a Waku node public [listening](#transport) and [discovery](#peer-discovery) address.
```
Comment (fryorcraken) [inline docs/overview/reference/glossary.md:63]: It used to generate the Peer id of the node. Which in turns, is present in the multiaddr of the node that is used by others to connect to said node.

Finally, this is a security measure as authentication method is used by libp2p at connection time to verify that a node is truly who they are based on their peer id.

File: scripts/config-table-generator.py [hunk 12]
```diff
@@ -0,0 +1,83 @@
+import re
```
Comment (fryorcraken) [inline scripts/config-table-generator.py:1]: Where is this used?

**Review by fryorcraken — APPROVED** [review]
Great job!

File: docs/guides/nwaku/run-docker.md (hunk 7 above)
Comment (LordGhostX) [reply docs/guides/nwaku/run-docker.md:12]: I tried it out, didn't work on Mac. I'll remove that line

File: scripts/config-table-generator.py (hunk 12 above)
Comment (LordGhostX) [reply scripts/config-table-generator.py:1]: I wrote this to automatically generate the markdown tables for the [Node Configuration Options](https://github.com/waku-org/docs.waku.org/blob/add-nwaku-guides/docs/guides/reference/node-config-options.md) page from the [wakunode2 config file](https://github.com/waku-org/nwaku/blob/master/apps/wakunode2/config.nim)

File: docs/guides/nwaku/build-source.md (hunk 6 above)
Comment (LordGhostX) [reply docs/guides/nwaku/build-source.md:82]: Both of them work fine, I'll just use `make update` in the docs

File: docs/guides/reference/node-config-options.md [hunk 13]
```diff
@@ -0,0 +1,165 @@
+---
+title: Node Configuration Options
```
Comment (jm-clius) [inline docs/guides/reference/node-config-options.md:2]: This is a great page!

**Review by jm-clius — COMMENTED** [review]
Great work! Apologies, I had some pending comments but only had chance to finish my review now. I know it's already merged, but feel free to use my comments in upcoming PRs as you see fit.

File: docs/guides/nwaku/build-source.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+To learn more about running nwaku, please refer to:
+
+- [Run a Nwaku Node](/guides/run-nwaku-node#run-the-node)
+- [Run Nwaku in a Docker Container](/guides/nwaku/run-docker)
+- [Run Nwaku with Docker Compose](/guides/nwaku/run-docker-compose)
+- [Node Configuration Methods](/guides/reference/node-config-methods)
+
+## Run Test Suite
+
+Run the tests for both `Waku v1` and `Waku v2`:
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply docs/guides/nwaku/build-source.md:101]: Good point from @jm-clius there. Maybe usage of v2 vs v1 is to be reviewed across the doc @LordGhostX

**Pull request merged**